--- /dev/fd/63	2026-04-27 16:21:35.030081132 +0400
+++ atom_vec_sph.cpp	2026-04-27 16:21:35.019996300 +0400
@@ -32,13 +32,22 @@
   atom->cv_flag = 1;
   atom->vest_flag = 1;
 
+  atom->stfluid_flag = 1;
+  atom->stmic_flag = 1;
+  atom->gradv_flag = 1;
+  atom->gammadot_flag = 1;
+  atom->fintx_flag = 1;
+  atom->finty_flag = 1;
+
   // strings with peratom variables to include in each AtomVec method
   // strings cannot contain fields in corresponding AtomVec default strings
   // order of fields in a string does not matter
   // except: fields_data_atom & fields_data_vel must match data file
 
-  fields_grow = {"rho", "drho", "esph", "desph", "cv", "vest"};
-  fields_copy = {"rho", "drho", "esph", "desph", "cv", "vest"};
+  fields_grow = {"rho", "drho", "esph", "desph", "cv", "vest",
+                 "stfluid", "stmic", "gradv", "gammadot", "fintx", "finty"};
+  fields_copy = {"rho", "drho", "esph", "desph", "cv", "vest",
+                 "stfluid", "stmic", "gradv", "gammadot", "fintx", "finty"};
   fields_comm = {"rho", "esph", "vest"};
   fields_comm_vel = {"rho", "esph", "vest"};
   fields_reverse = {"drho", "desph"};
@@ -46,7 +55,8 @@
   fields_border_vel = {"rho", "esph", "cv", "vest"};
   fields_exchange = {"rho", "esph", "cv", "vest"};
   fields_restart = {"rho", "esph", "cv", "vest"};
-  fields_create = {"rho", "esph", "cv", "vest", "desph", "drho"};
+  fields_create = {"rho", "esph", "cv", "vest", "desph", "drho",
+                   "stfluid", "stmic", "gradv", "gammadot", "fintx", "finty"};
   fields_data_atom = {"id", "type", "rho", "esph", "cv", "x"};
   fields_data_vel = {"id", "v"};
 
